### DISCUSSION TITLE
How to get around "the rule matches the empty string"?

### DISCUSSION THREAD

**Discussion by ethindp** [tree-sitter/tree-sitter#2527] [Q&A] [answered]
I'm writing an Ada compiler and there are a few rules that match the empty string. What's a good workaround for this, given tree-sitter doesn't support it unless it's the start rule? (I could just directly inline the rule, but that doesn't seem like the most efficient solution.)

**Comment by ahlinc** [comment]
Could you specify what do you mean by _the empty string_? An empty string literal like `""`?

**Comment by ethindp** [comment]
@ahlinc No, the Ada grammar has rules like `declarative_part` which consists of zero or more `declarative_item`s. So tree-sitter is complaining because that rule (`declarative_part: $ => repeat($.declarative_item)`) matches nothing (the empty string) as input.

**Comment by amaanq** [comment, marked as answer]
why not just make it repeat1 and wherever it's used in another rule, marked as optional?

**Reply by ahlinc** [reply]
That's why I asked my previous question to get a clarity that it doesn't related to terminals.
